mlrun/secrets.py

```diff
@@ -161,6 +192,9 @@ def get_secret_or_env(
     4. An MLRun-generated env. variable, mounted from a project secret (to be used in MLRun runtimes)
     5. The default value
 
+    Also supports discovering the value inside any environment variable that contains a JSON-encoded list
+    of dicts with fields: {'name': 'KEY', 'value': 'VAL', 'value_from': ...}. This fallback is applied
+    after checking normal environment variables and before returning the default.
```

Comment:
why not supporting `V1EnvVarSource` object as well then? as on the client side we use this type to set env from secret:

https://github.com/mlrun/mlrun/blob/c36122fc31d3870a252575a3019e26ae8128b24a/mlrun/runtimes/pod.py#L711-L717

And could you please add the description what is the general motivation for supporting dict here?